test/parallel/test-worker-cleanexit-with-moduleload.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+require('../common');
+
+// Harden the thread interactions on the exit path.
+// Ensure workers are able to bail out safe at
+// arbitrary execution points. By using a number of
+// internal modules as load candidates, the expectation
+// is that those will be at various control flow points
+// preferrably in the C++ land.
+
+const { Worker } = require('worker_threads');
+for (let i = 0; i < 10; i++) {
+  new Worker("const modules = ['fs', 'assert', 'async_hooks'," +
+    "'buffer', 'child_process', 'net', 'http', 'https', 'os'," +
+    "'path', 'v8', 'vm'];" +
+    'modules.forEach((module) => {' +
+    'const m = require(module);' +
+    '});', { eval: true });
+}
+
+// Allow workers to go live.
+setTimeout(() => {
+  process.exit(0);
+}, 200);
```

Comment:
Would it make sense to use `common.platformTimeout()` here or is there no real chance that a very slow machine will bail before the Worker gets where we want it to be?

Reply:
@Trott - that is a very interesting question.

If the host has slow network (which I believe the `common.platformTimeout` controls), it should not affect the timing according to me, as the workers are doing some CPU bound stuff.

If we lower the timeout value, we run the risk of workers not yet entered into the space we want those to.
If we raise the timeout value, we run the risk of workers already exited the space we want those to.

Clearly what we want is most of the workers in the thick of module loading activities.

In short, though the current value may not be the best, it serves the purpose of leaving the workers at different program stages, and if that varies based on platforms, that is what we desire - in expectation of getting more variations.

thoughts?

Reply:
You're definitely more informed on what the test is trying to do and all the surrounding considerations than I am, so I defer to your judgment on this. Just wanted to make sure it had been considered. I have no opinion one way or the other.